src/ui/ui_exports.js

```diff
@@ -37,22 +43,43 @@ class UiExports {
     this.consumers.push(consumer);
   }
 
+  // when module ids are defined in a uiExports spec they can
+  // be defined as relative to the public directory
+  resolveModulePath(plugin, moduleId) {
+    const isRelative = moduleId.startsWith('./') || moduleId.startsWith('.\\');
+    if (isRelative) return join(`plugins/${plugin.id}`, moduleId);
+    return moduleId;
+  }
+
+  extendAliases(plugin, alias, idOrIds) {
+    const existingModuleIds = this.aliases[alias] || [];
+    const resolve = id => this.resolveModulePath(plugin, id);
+    const resolvedModuleIds = arr(idOrIds).map(resolve);
+
+    this.aliases[alias] = _.union(existingModuleIds, resolvedModuleIds);
+  }
+
   exportConsumer(type) {
-    for (let consumer of this.consumers) {
+    for (const consumer of this.consumers) {
       if (!consumer.exportConsumer) continue;
-      let fn = consumer.exportConsumer(type);
+      const fn = consumer.exportConsumer(type);
       if (fn) return fn;
     }
 
     switch (type) {
-      case 'app':
       case 'apps':
-        return (plugin, specs) => {
-          for (let spec of [].concat(specs || [])) {
-            let app = this.apps.new(_.defaults({}, spec, {
+        return (plugin, spec) => {
+          for (const rawSpec of arr(spec)) {
```

Comment:
the function that is returned on line 71 is fed the absolute value of anything found at either `apps` or `app` keys in uiExports. Nearly every plugin currently exports a single app as 
```js
new kibana.Plugin({
  ui: {
    app: {
      //..
    }
  }
});
```

But then there is sense, it exports two apps: https://github.com/elastic/sense/blob/master/index.js#L112